addons/mail/static/src/core/common/thread_model.js

```diff
@@ -526,28 +526,36 @@ export class Thread extends Record {
         return this.isChatChannel ? "@" : "#";
     }
 
-    get lastSelfMessageSeenByEveryone() {
-        const otherMembers = this.channelMembers.filter((member) =>
-            member.persona.notEq(this.store.self)
-        );
-        if (otherMembers.length === 0) {
-            return false;
-        }
-        const otherLastSeenMessageIds = otherMembers
-            .filter((member) => member.seen_message_id)
-            .map((member) => member.seen_message_id.id);
-        if (otherLastSeenMessageIds.length === 0) {
-            return false;
-        }
-        const lastMessageSeenByAllId = Math.min(...otherLastSeenMessageIds);
-        const orderedSelfSeenMessages = this.persistentMessages.filter((message) => {
-            return message.author?.eq(this.store.self) && message.id <= lastMessageSeenByAllId;
-        });
-        if (!orderedSelfSeenMessages || orderedSelfSeenMessages.length === 0) {
-            return false;
-        }
-        return orderedSelfSeenMessages.slice().pop();
-    }
+    lastSelfMessageSeenByEveryone = Record.one("Message", {
+        compute() {
+            const otherMembers = this.channelMembers.filter((member) =>
```

Comment:
I believe we should also have a quick return to not compute anything for non-channel models and for channel_type channel. (all those with the seen feature disabled)